examples/config.json

```diff
@@ -149,7 +149,7 @@
     },
 
     "ornl.summit_prte" : {
-        "project"  : "CSC343",
+        "project"  : "geo111",
```

Comment:
I think we should also depart from hard-coding project IDs into our example configs. Should we just pass it with a env and an exception if it is not defined?

Reply:
Ha, this surprises me! *you* suggest an env variable (I never though I'd see the day), and *me* saying `nah, not really` - whats wrong?  Is the world coming to an end?? :-)

In all sincerity though, I think env vars are great for settings which vary from run to run, and config files are great for settings which vary from user to user or resource to resource.

Having said that, our `examples/config.json` is clearly not a satisfactory solution.  Apart from the fact that I should not have committed the above, its obvious that it needs to be changed for every user on every tutorial.

I would envision a better solution to have something like a `~/.radical/pilot/examples.cfg` with very much the same settings which we have in the current example config, so that those data are specific to users and tutorials, and only fall back to `examples/examples.cfg` if the above does not exist.  We can then document that behavior in `examples/examples.cfg` and prepare it as a template for the user to place into `~/.radical` as part of the tutorial setup.  Would something like this makes sense IYO, or is it overkill / off target?  I wouldn't mind having a go at it...

Reply:
I think about examples as something a new user wants to try out as sort of RP hello world. I would use the simplest, most self evident possible approach. To me, having to know about, and write or edit my own `~/.radical/pilot/examples.cf` does not seem to fit that requirement. Maybe I am wrong in my approach to the whole examples thing though.

Reply:
We can check what other systems are doing.  If I'm not mistaken, many do require changing code or config files before running any example.  FWIW, so does EnTK :-)

A really self sufficient option would be to get rid of the config altogether, and have a small section at the beginning of each example, like

```python
#!/usr/bin/env python

# configuration for target resource
RP_RESOURCE = 'local.localhost'
RP_SCHEMA = 'fork'
RP_QUEUE = None
...
```

That is just a few lines, and easy to change and to reference in the documentation.  We can still document our config system in one of the advanced examples (if we want).

Not saying that this is the best way, but it is a rather simple and self-sufficient option.

Reply:
I moved this discussion to #1989